### PR TITLE
Reduce usage of testUsingContext

### PR DESCRIPTION
**Pull request by christopherfujino**
Part of https://github.com/flutter/flutter/issues/47161

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/cmake_test.dart
Comment (eliasyishak) [inline packages/flutter_tools/test/general.shard/cmake_test.dart:43]: Were these overrides just being provided because they were being copied from a previous test? I don't see anywhere in this PR where we are injecting the filesystem and process manager

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/cmake_test.dart:43]: yeah, only a few of these tests actually hit the context

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/cmake_test.dart:43]: As a side note, I'm exploring writing a script that will test what happens if we rename each `testUsingContext` test as `testWithoutContext` and see how many still pass.

**Review by eliasyishak — APPROVED** [review]
lgtm, lmk if you want to do another review if you get the script to replace `testUsingContext` calls

**Comment by christopherfujino** [comment]
> lgtm, lmk if you want to do another review if you get the script to replace `testUsingContext` calls

For clarity, that would be a new PR. For this change, I essentially manually did this for the test libraries I changed.

**Pull request merged**
